lua/entities/gmod_wire_expression2/core/wirelink.lua

```diff
@@ -635,3 +635,100 @@ e2function number wirelink:writeTableSimple(address, table data)
 	if not validWirelink(self, this) or not this.WriteCell then return 0 end
 	return writeArraySimple(this, address, data.n)
 end
+
+-- Events for when THIS e2 is read or written to via wirelink, hispeed, whatever uses read & write cell.
+
+-- Events can't have return values, lambdas can't have multiple return types
+-- Don't want to risk a magic number as error causing a collision, so I figure
+-- functions for returning hispeed error & value are the best compromise.
+
+__e2setcost(2)
+e2function void returnHiSpeedValue(number value)
+	self.data.hispeedIOError = false
+	self.data.readCellValue = value
+end
+
+e2function void setHiSpeedError(number value)
+	self.data.hispeedIOError = value ~= 0
+end
+
+e2function void returnHiSpeedError()
+	self.data.hispeedIOError = true
```

Comment:
The function name should start with `hispeed`: `hispeedSetValue` `hispeedSetError`. What's the point of the `self.data.hispeedIOError = true` one?

Reply:
> What's the point of the `self.data.hispeedIOError = true` one?

I initially had only `returnHiSpeedError` thinking that you should only call the func as the last thing before exiting, but having `returnHiSpeedValue` as the only option to clear it might lead to people doing silly things so I added `setHiSpeedError` as an alternative